dace/codegen/targets/intel_fpga.py

```diff
@@ -0,0 +1,1147 @@
+        defined_symbols.update({k: v.dtype for k, v in sdfg.constants.items()})
+
+        for connector, (memlet, _, _) in memlets.items():
+            if connector is not None:
+                defined_symbols.update({connector: sdfg.arrays[memlet.data].dtype})
+
+        used_streams = []
+        for stmt in body:  # for each statement in tasklet body
+            if isinstance(stmt, ast.Expr):
+                ocl_visitor = OpenCLDaceKeywordRemover(sdfg, memlets, sdfg.constants)
+                rk = ocl_visitor.visit_TopLevelExpr(stmt)
+            else:
+                ocl_visitor = OpenCLDaceKeywordRemover(sdfg, self._dispatcher.defined_vars, memlets, sdfg.constants)
+                rk = ocl_visitor.visit(stmt)
+            used_streams.extend(ocl_visitor.used_streams)
+            if rk is not None:
+                result = StringIO()
+                cppunparse.CPPUnparser(rk, ldepth + 1, locals, result,
+                                       defined_symbols=defined_symbols, type_inference=True)
+                callsite_stream.write(result.getvalue(), sdfg, state_id, node)
+
+        # In Intel OpenCL is not possible to have multiple access points to the same channel
+        for s in used_streams:
+            if used_streams.count(s) > 1:
+                raise dace.codegen.codegen.CodegenError(
+                    "Multiple access points for stream are forbidden in IntelFPGA (stream \"{}\" "
+                    "in tasklet \"{}\")".format(s, node.name))
+
+    def generate_constants(self, sdfg, callsite_stream):
+        # Write constants
+        for name, val in sdfg.constants.items():
+            if isinstance(val, np.ndarray):
+                if isinstance(val, ndarray.ndarray):
+                    dtype = val.descriptor.dtype
+                else:
+                    dtype = dtypes.typeclass(val.dtype.type)
+                const_str = "__constant " + dtype.ctype + \
+                            " " + name + "[" + str(val.size) + "] = {"
+                it = np.nditer(val, order='C')
+                for i in range(val.size - 1):
+                    const_str += str(it[0]) + ", "
+                    it.iternext()
+                const_str += str(it[0]) + "};\n"
+                callsite_stream.write(const_str, sdfg)
+            else:
+                callsite_stream.write(
+                    "__constant %s %s = %s;\n" % (dtypes._CTYPES[type(val)],
+                                                  name, str(val)), sdfg)
+
+
+class OpenCLDaceKeywordRemover(cpu.DaCeKeywordRemover):
+    """
+    Removes Dace Keywords and Enforce OpenCL compliance
+    """
+
+    def __init__(self, sdfg, defined_vars, memlets, *args, **kwargs):
+        self.sdfg = sdfg
+        self.defined_vars = defined_vars
+        self._nptypes_to_ctypes = {'float64': 'double'}
+        self._nptypes = ['float64']
+        self._ctypes = ['bool', 'char', 'cl_char', 'unsigned char', 'uchar', 'cl_uchar', 'short', 'cl_short',
```

Comment:
These can be defined for the class instead of per-object

Reply:
Done